### PR TITLE
Frontend event callbacks through js events. Getter support from frontend API.

### PR DESCRIPTION
**Pull request by DDRBoxman**
I've had this sitting around on my computer for a while now. I figured I should just push it and get some feedback.

I also fixed the framework path issue on OSX with newer CEF versions.

Addresses at least in part:
#50 #38 #35

### CODE REVIEW

File: cef-isolation/main.mm [hunk 1]
```diff
@@ -86,6 +86,7 @@ int main (int argc, const char * argv[])
 		
 		settings.log_severity = LOGSEVERITY_DEFAULT;
 		settings.windowless_rendering_enabled = true;
+		settings.remote_debugging_port = 2012;
```
Comment (OsirisNL) [inline cef-isolation/main.mm]: Remote debugging should probably not be enabled by default.

File: shared/browser-app.cpp [hunk 2]
```diff
@@ -105,11 +108,45 @@ bool BrowserApp::OnProcessMessageReceived(CefRefPtr<CefBrowser> browser,
...
+			jsonString);
+
+		free(jsonString);
+
+		CefRefPtr<CefV8Value> returnValue;
+		CefRefPtr<CefV8Exception> exception;
+
+		// Create the CustomEvent object
+		// We have to use eval to invoke the new operator
+		context->Eval(script, returnValue, exception);
```
Comment (OsirisNL) [inline shared/browser-app.cpp]: This method's arguments have been changed in CEF branch 2840, should now be:

`context->Eval(script, browser->GetMainFrame()->GetURL() ,0, returnValue, exception);`

**Comment by RytoEX** [comment]
There's a typo in the commit message and PR title ("Frontened").

**Comment by DDRBoxman** [comment]
I'm spending the time to also go through and add getters to some of the frontend API methods that relate.

**Comment by OsirisNL** [comment]
Are there any more commits coming, or can I merge this?

**Comment by DDRBoxman** [comment]
Those last couple should wrap it up! :shipit:

**Comment by Lange** [comment]
Just wanted to chime in and say how thankful I am for all the recent activity on obs-browser. It's very exciting to see these things getting added that I've wanted for so long. @DDRBoxman & @OsirisNL thank you both so much!

**Comment by theunknownartisthour** [comment]
Anyone in charge of obs-studio so that we can get the changes to the submodule moved up until this point?

**Comment by btorellALTA** [comment]
The submodule hasn't been updated, but the latest release of OBS Studio should include these changes. The full OBS Studio installer should include the 1.27 release of obs-browser bundled.

**Comment by SReject** [comment]
Could we get an update to README.md documenting the changes to the JS API?  

Or even better, a JSAPI.md doc detailing it.

**Comment by OsirisNL** [comment]
Everything should already be in there

**Pull request merged**